src/main/java/com/commercetools/emailprocessor/email/EmailProcessor.java

```diff
@@ -89,9 +94,10 @@ int callApiEndpoint(final String customObjectId, final TenantConfiguration tenan
         try {
             HttpPost httpPost = tenantConfiguration.getHttpPost();
             List<NameValuePair> params = new ArrayList<NameValuePair>();
-            params.add(new BasicNameValuePair(PARAM_EMAIL_ID, customObjectId));
+            params.add(new BasicNameValuePair(PARAM_EMAIL_ID, encrypt(customObjectId, tenantConfiguration
+                .getEncryptionKey(), Cipher.ENCRYPT_MODE)));
             params.add(new BasicNameValuePair(PARAM_TENANT_ID, tenantConfiguration.getProjectKey()));
-            httpPost.setEntity(new UrlEncodedFormEntity(params));
+            httpPost.setEntity(new UrlEncodedFormEntity(params, Charset.defaultCharset()));
             response = HttpClients.createDefault().execute(httpPost);
             responseCode = response.getStatusLine() != null ? response.getStatusLine().getStatusCode() : Statistics
```

Comment:
initializing now the responseCode is redundant because it will always be overwritten here.

Reply:
b143f2b